src/compiler/transformers/es2020.ts

```diff
@@ -54,7 +54,7 @@ namespace ts {
                 Debug.assertNotNode(chain, isNonNullChain);
                 links.unshift(chain);
             }
-            return { expression: chain.expression, chain: links };
+            return { expression: skipPartiallyEmittedExpressions(chain.expression), chain: links };

```

Comment:
This will result in a loss of some leading or trailing comments. I'd recommend doing this in `visitOptionalExpression` instead, and using `factory.restoreOuterExpressions` to preserve comments:

```ts
// visitOptionalExpression
const left = visitNonOptionalExpression(skipPartiallyEmittedExpressions(expression), isCallChain(chain[0]), /*isDelete*/ false);
//                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- Skip here
const leftThisArg = isSyntheticReference(left) ? left.thisArg : undefined;
let capturedLeft = isSyntheticReference(left) ? left.expression : left;
//  ^^^^^^^^^^^^- Swap assignment order for capturedLeft/leftExpression
let leftExpression = factory.restoreOuterExpressions(expression, capturedLeft, OuterExpressionKinds.PartiallyEmittedExpressions);
//                           ^^^^^^^^^^^^^^^^^^^^^^^- Restore here
if (!isSimpleCopiableExpression(capturedLeft)) {
    //                          ^^^^^^^^^^^^- Swap argument to capturedLeft
    capturedLeft = factory.createTempVariable(hoistVariableDeclaration);
    leftExpression = factory.createAssignment(capturedLeft, leftExpression);
}
```

Reply:
I would also add a test for comment preservation, something like this:
```ts
/*a1*/(/*a2*/foo.m as any/*a3*/)/*a4*/?.();
/*b1*/(/*b2*/<any>foo.m/*b3*/)/*b4*/?.();
```

Reply:
It seems with both options comments `/*a2*/` and `/*b2*/` will be lost. Will check that case 
```js
// return { expression: skipPartiallyEmittedExpressions(chain.expression), chain: links };

/*a1*/ foo.m /*a3*/ /*a4*/?.();
/*b1*/ foo.m /*b3*/ /*b4*/?.();
````

```js
// factory.restoreOuterExpressions
/*a1*/ foo.m /*a3*/ /*a4*/?.();
/*b1*/ foo.m /*b3*/ /*b4*/?.();
```

Reply:
@rbuckton Thanks for the feedback. How TS should emit comments for that case? 

```js
// es2015
/*a1*/ (_c = foo.m /*a3*/ /*a4*/) === null || _c === void 0 ? void 0 : _c.call(/*a2*/ foo);
/*b1*/ (_d = foo.m /*b3*/ /*b4*/) === null || _d === void 0 ? void 0 : _d.call(foo);
```

```js
// esnext
/*a1*/ foo.m /*a3*/ /*a4*/?.(); 
/*b1*/ foo.m /*b3*/ /*b4*/?.();
```
Do we need to attach `/*a2*/`/`/*b2*/` after `/*a1*/`/`/*b1*/`?

Reply:
For both I would expect `/*a2*/`/`/*b2*/` to be attached after `/*a1*/`/`/*b1*/`, however the fact those comments aren't present in `esnext` is an issue with how `PartiallyEmittedExpression` nodes are handled in the emitter. I can push a fix to this PR that handles that and this should be good to merge.